src/hiro_graph_client/client.py

```diff
@@ -247,38 +219,35 @@ def get_timeseries(self, node_id: str, starttime: str = None, endtime: str = Non
         }
 
         url = self._endpoint + '/' + quote_plus(node_id) + '/values' + self._get_query_part(query)
-        res = self.get(url, token=token)
+        res = self.get(url)
         if 'error' in res:
             return res
         timeseries = res['items']
         timeseries.sort(key=lambda x: x['timestamp'])
         return timeseries
 
-    def post_timeseries(self, node_id: str, items: list, token: str = None) -> dict:
+    def post_timeseries(self, node_id: str, items: list) -> dict:
         """
         HIRO REST query API: `POST self._endpoint + '/{id}/values'`
 
         :param node_id: ogit/_id of the node containing timeseries
         :param items: list of timeseries values [{timestamp: (ms since epoch), value: ...},...]
-        :param token: Optional external token.
         :return: The result payload
         """
         url = self._endpoint + '/' + quote_plus(node_id) + '/values?synchronous=true'
         data = {"items": items}
-        return self.post(url, data, token=token)
+        return self.post(url, data)
 
     def get_attachment(self,
                        node_id: str,
                        content_id: str = None,
-                       include_deleted: bool = False,
-                       token: str = None) -> Iterator[bytes]:
+                       include_deleted: bool = False) -> Iterator[bytes]:
```

Comment:
prefer a generator over an iterator due to it's context preserving nature

Reply:
This is the right Typing for a Generator that only returns values and does not receives any. See https://docs.python.org/3/library/typing.html#typing.Generator

Reply:
Although i should yield the values returned from the root iterator in get_binary and not return them. I will change this accordingly.

Reply:
Done